### PR TITLE
Update high-availability.md - Fix value of kubernetesVersion in the kubeadm-config.yaml

### PR DESCRIPTION
**Pull request by ocofaigh**
Fix for https://github.com/kubernetes/website/issues/9594#

Changed the value of `kubernetesVersion` to `v1.11.1` in the kubeadm-config.yaml so it pulls the correct server images for the latest v1.11 release.

I don't like the idea of having to change this value for every point release. Maybe someone can come up with a better solution than submitting a change to the documentation for every minor release?

### CODE REVIEW

**Comment by ocofaigh** [comment]
I believe I have requested CLA access. Can someone confirm please?

**Comment by neolit123** [comment]
@ocofaigh thanks for the PR. the system still says that you haven't signed the CLA.
if you have a problem with that, please email `info at cncf.io`.

**Comment by ocofaigh** [comment]
CLA signed. I got confirmation email

**Comment by neolit123** [comment]
this looks good to me, but

> I don't like the idea of having to change this value for every point release. Maybe someone can come up with a better solution than submitting a change to the documentation for every minor release?

i don't like the idea either. i you can override the config version using `--kubernetes-version=1.11.x`, but it's a question if we should bump this every time.

waiting on comments from:
/cc @kubernetes/sig-cluster-lifecycle-pr-reviews

**Comment by chuckha** [comment]
That configuration is already a template. Maybe we could advise the reader to substitute the correct patch version and give an example if they don't feel like looking up the newest patch version? Then we could say "v1.11.x please replace x".

**Comment by neolit123** [comment]
> That configuration is already a template. Maybe we could advise the reader to substitute the correct patch version and give an example if they don't feel like looking up the newest patch version? Then we could say "v1.11.x please replace x".

i second that.

@ocofaigh, please modify the example config version to `1.11.x` and add a note on top about it.

**Comment by ocofaigh** [comment]
Changes made as per above comments

**Comment by neolit123** [comment]
@ocofaigh 
thanks for the update, this looks good to me.

**Comment by mistyhacks** [comment]
I've added this so the agenda for the next SIG-Docs meeting. I am not sure this is the best end-state to solve this problem.

/hold

**Comment by neolit123** [comment]
@mistyhacks 
> I've added this so the agenda for the next SIG-Docs meeting. I am not sure this is the best end-state to solve this problem.

was this discussed at the sig meeting?

**Comment by mistyhacks** [comment]
Yes, but we haven't yet decided on the best approach. /cc @zacharysarah

**Comment by zacharysarah** [comment]
@ocofaigh Thanks for waiting. 

TLDR: this PR is approved. 👍 

@neolit123 Are updates for point releases sufficiently bothersome that automation is worth exploring? Or is it just a routine (if annoying) part of ongoing maintenance? It could be cool to automate version as a variable, but I'm concerned that:

1. The scope of changes would be large.

Implementing a site-wide variable requires non-trivial work. It's easy to find wherever the docs mention a specific version, but determining whether an instance is contextually appropriate to automate would probably require manual review of each instance.

2. Automated updates might increment version incorrectly.

It's probably a minor concern for point releases, but which is worse: an unknown state (out of date docs don't indicate whether a newer version is supported) or an incorrect state (the docs incorrectly state version support)? This may not be relevant, and I'm happy to withdraw my concern if it's out of scope to the question.

@neolit123 What do you think?

/cc @mistyhacks 

/lgtm
/approve

**Comment by neolit123** [comment]
@zacharysarah 

thank you for bringing this topic.

> Are updates for point releases sufficiently bothersome that automation is worth exploring? Or is it just a routine (if annoying) part of ongoing maintenance?

i personally do not like the `1.11.x` solution that this PR adds.
i think we should automate the example configs as much as possible across `website`.

> It could be cool to automate version as a variable, but I'm concerned that:

> The scope of changes would be large.
Implementing a site-wide variable requires non-trivial work. It's easy to find wherever the docs mention a specific version, but determining whether an instance is contextually appropriate to automate would probably require manual review of each instance.

this type of a variable can be useful but probably only once the kubeadm config is GA, as currently it is still in alpha and ever changing. this makes it so that it needs manual updates and by doing manual updates we can queue version changes as well.

> Automated updates might increment version incorrectly.
It's probably a minor concern for point releases, but which is worse: an unknown state (out of date docs don't indicate whether a newer version is supported) or an incorrect state (the docs incorrectly state version support)? This may not be relevant, and I'm happy to withdraw my concern if it's out of scope to the question.

i think ideally these example configs should be auto generated and embedded as `.mds`. this way they can include the latest k8s version and the same time be valid as they are supposedly tool generated.

one big problem with this solution is that `k/website` does not update the generated documentation on each PATCH release, but only on MINOR releases.

has this been considered at some point in time?

**Comment by zacharysarah** [comment]
> one big problem with this solution is that k/website does not update the generated documentation on each PATCH release, but only on MINOR releases.
> has this been considered at some point in time?

We have [instructions for regenerating docs](https://kubernetes.io/docs/contribute/generate-ref-docs/kubernetes-api/). We'd certainly welcome PRs for minor release updates! 👍
